File: translate_api.php
<?php
if(isset($_POST['text']) && isset($_POST['target_language'])) {
    $input_text = $_POST['text'];
    $target_language = $_POST['target_language'];

    // Your translation API endpoint URL
    $url = 'http://localhost:5000/translate';
    
    // Data to be sent in the POST request
    $data = array(
        'inputText' => $input_text,
        'targetLanguage' => $target_language
    );

    // Constructing HTTP POST request
    $options = array(
        'http' => array(
            'header'  => "Content-type: application/x-www-form-urlencoded\r\n",
            'method'  => 'POST',
            'content' => http_build_query($data),
        ),
    );
    
    // Creating a stream context for the HTTP request
    $context  = stream_context_create($options);
    
    // Making the HTTP POST request
    $result = file_get_contents($url, false, $context);
    
    // Checking if the request was successful
    if ($result === FALSE) {
        echo "Error: Unable to connect to the translation service.";
    } else {
        // Parsing the JSON response
        $json = json_decode($result, true);
        
        // Checking if the translation was successful
        if (isset($json['translatedText'])) {
            echo $json['translatedText'];
        } else {
            echo "Error: Translation failed.";
        }
    }
} else {
    echo "Error: Input text or target language not provided.";
}
?>

File: translate_api.py
from flask import Flask, request, jsonify
from easy_google_translate import EasyGoogleTranslate
from flask_cors import CORS

app = Flask(__name__)
CORS(app)  # Enable CORS for all routes
translator = EasyGoogleTranslate()

@app.route('/translate', methods=['POST'])
def translate():
    # Ensure that the required data is present in the request
    if 'text' not in request.form or 'target_language' not in request.form or 'detected_language' not in request.form:
        return jsonify({'error': 'Incomplete request. Required parameters are missing.'}), 400
    
    input_text = request.form['text']
    target_language = request.form['target_language']
    detected_language = request.form['detected_language']

    try:
        # Translate the input text
        translated_text = translator.translate(input_text, source_language=detected_language, target_language=target_language)
        # Construct the response JSON object with translation information
        response = {
            'inputText': input_text,
            'detectedLanguage': detected_language,
            'targetLanguage': target_language,
            'translatedText': translated_text
        }
        return jsonify(response), 200
    except Exception as e:
        return jsonify({'error': str(e)}), 500

if __name__ == '__main__':
    app.run(debug=True)
